images/pytorch-pods/entrypoint.sh

```diff
@@ -13,24 +13,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
 source /setup.sh
 
+set -u
+set -x
 
-# TODO: Make these arguments piped through from jsonnet
-export MACHINE_TYPE=n1-standard-16
-export ACCELERATOR_TYPE=v3-32
-export RUNTIME_VERSION=pytorch-nightly
 export RESOURCE_SUFFIX=$POD_UID
-source /setup-pytorch-pods.sh
+export INSTANCE_TEMPLATE_NAME="instance-template-${RESOURCE_SUFFIX}"
+export TPU_POD_NAME="tpu-pod-${RESOURCE_SUFFIX}"
+export INSTANCE_GROUP_NAME="instance-group-${RESOURCE_SUFFIX}"
 
-set -u
-set -x
+# zone/name -> name
+export TPU_POD_NAME=$(echo ${TPU_NAME} | awk -F'/' '{print $2}')
 
+# HACK: create symlink where other images have `gcloud` installed
+mkdir -p /root/google-cloud-sdk/bin/
+ln -s $(which gcloud) /root/google-cloud-sdk/bin/gcloud
 source /publish.sh
 
-# "$@" should look like python -m torch_xla.distributed.xla_dist --tpu=...
-docker-entrypoint.sh "$@"
-
-# Teardown resources
-/teardown-pytorch-pods.sh
+timeout 180 /setup-instances.sh && \
+export master=$(gcloud compute instance-groups \
+  list-instances \
+  ${INSTANCE_GROUP_NAME} \
+  --zone=${ZONE} \
+  --format="value(NAME)" \
+  --limit=1) && \
+gcloud -q compute ssh --internal-ip --zone=$ZONE $master \
+  --command "source /anaconda3/etc/profile.d/conda.sh && \
+  conda activate torch-xla-nightly && \
+  python -m torch_xla.distributed.xla_dist --tpu=$TPU_POD_NAME --conda-env=torch-xla-nightly -- $*"
```

Comment:
This way for PT/TPU Pods tests jsonnet configs, we're only able to specify the `python $SCRIPT $ARGS` without the `xla_dist` module flags? I think we need the ability to specify other flags like `--env` (it's an `xla_dist` module flag) and also switch out the `--conda-env` versions to like `torch-xla-1.5` or smth.

Can we build the `python -m torch_xla...` full command from jsonnet?

Reply:
Added template parameters for conda env and `xla_dist` flags. [oneshot link](https://pantheon.corp.google.com/kubernetes/job/europe-west4-a/europe-oneshots/default/pt-nightly-mnist-convergence-v2-32-nfnsj?project=xl-ml-test&tab=details&duration=P30D&pod_summary_list_tablesize=20&service_list_datatablesize=20)